monai/losses/shape.py

```diff
@@ -0,0 +1,172 @@
+                - ``"sum"``: the output will be summed.
+
+        Raises:
+            TypeError: When ``other_act`` is not an ``Optional[Callable]``.
+            ValueError: When more than 1 of [``sigmoid=True``, ``softmax=True``, ``other_act is not None``].
+                Incompatible values.
+        """
+        super().__init__(reduction=LossReduction(reduction).value)
+        if other_act is not None and not callable(other_act):
+            raise TypeError(f"other_act must be None or callable but is {type(other_act).__name__}.")
+        if int(sigmoid) + int(softmax) + int(other_act is not None) > 1:
+            raise ValueError("Incompatible values: more than 1 of [sigmoid=True, softmax=True, other_act is not None].")
+        self.include_background = include_background
+        self.to_onehot_y = to_onehot_y
+        self.sigmoid = sigmoid
+        self.softmax = softmax
+        self.other_act = other_act
+        self.smooth_nr = float(smooth_nr)
+        self.smooth_k = float(smooth_k)
+
+    def distance_map(self, mask: torch.Tensor) -> torch.Tensor:
+        """Creates a distance map from a 2D binary mask.
+
+        Args:
+            mask (torch.Tensor): Binary mask of shape [WD].
+
+        Returns:
+            torch.Tensor: Boundary distance map of shape [WD]
+        """
+        # Convert to NumPy to use with SciPy
+        roi: np.ndarray = mask.numpy()
+
+        # Compute normalized distance transform
+        dt: np.ndarray = ndimage.distance_transform_edt(roi)
+        dt /= dt.max() + self.smooth_nr
+
+        # apply Heaviside function to softly normalize into [0, 1]
+        result: np.ndarray = 1 / (1 + np.exp(-(1 - dt) / self.smooth_k))
+        # mask using region of interest
+        result *= roi
+
+        return torch.Tensor(result)
+
+    def forward(self, input: torch.Tensor, target: torch.Tensor) -> torch.Tensor:
+        """
+        Args:
+            input: the shape should be BNH[WD], where N is the number of classes.
+            target: the shape should be BNH[WD] or B1H[WD], where N is the number of classes.
+
+        Raises:
+            AssertionError: When input and target (after one hot transform if set)
+                have different shapes.
+            ValueError: When ``self.reduction`` is not one of ["mean", "sum", "none"].
+
+        Example:
+            >>> from monai.losses.dice import *  # NOQA
+            >>> import torch
+            >>> from monai.losses.dice import ShapeLoss
+            >>> B, C, H, W = 7, 5, 3, 2
+            >>> input = torch.rand(B, C, H, W)
+            >>> target_idx = torch.randint(low=0, high=C - 1, size=(B, H, W)).long()
+            >>> target = one_hot(target_idx[:, None, ...], num_classes=C)
+            >>> self = ShapeLoss(reduction='none')
+            >>> loss = self(input, target)
+            >>> assert np.broadcast_shapes(loss.shape, input.shape) == input.shape
```

Comment:
would be great to create a test case for this class, you can follow this one https://github.com/Project-MONAI/MONAI/blob/dev/tests/test_local_normalized_cross_correlation_loss.py

you can run a single test case within the codebase:
```bash
pip install -U -r requirements-dev.txt  # install testing tools
python -m tests.test_local_normalized_cross_correlation_loss
```